web/api/v1/api.go

```diff
@@ -255,7 +255,17 @@ func respond(w http.ResponseWriter, data interface{}) {
 
 func respondError(w http.ResponseWriter, apiErr *apiError, data interface{}) {
 	w.Header().Set("Content-Type", "application/json")
-	w.WriteHeader(422)
```

Comment:
422 is specifically for the WebDAV extension and isn't mentioned in any other HTTP rfc. The wording of 400 has been adjusted to make it clear that it isn't only about syntactically incorrect requests but any client error: https://tools.ietf.org/html/draft-ietf-httpbis-p2-semantics-26#section-6.5.1